tests/Test.MSAL.NET.Unit/PublicClientApplicationTests.cs

```diff
@@ -1049,5 +1049,66 @@ public void GetUserTest()
             Assert.AreEqual(userToFind.IdentityProvider, fetchedUser.IdentityProvider);
             Assert.AreEqual(userToFind.Name, fetchedUser.Name);
         }
+
+        [TestMethod]
+        [Description("Test for AcquireToken with user canceling authentication")]
+        public async Task AcquireTokenWithAuthenticationCanceledTestAsync()
+        {
+            PublicClientApplication app = new PublicClientApplication(TestConstants.ClientId);
+
+            // Interactive call and user cancels authentication
+            MockWebUI ui = new MockWebUI()
+            {
+                MockResult = new AuthorizationResult(AuthorizationStatus.UserCancel,
+                    TestConstants.AuthorityHomeTenant + "?error=user_canceled")
+            };
+
+            MockHelpers.ConfigureMockWebUI(ui);
+
+            try
+            {
+                AuthenticationResult result = await app.AcquireTokenAsync(TestConstants.Scope).ConfigureAwait(false);
+            }
+            catch (MsalClientException exc)
+            {
```

Comment:
If an exception is not thrown, your test will still pass. You can add a return statement in your catch block and an Assert.Fail("Should not reach here") at line 1082 (after the finally). Note that the finally clause is guaranteed to be executed, even if you return in catch block.

Reply:
cool...great suggestion. thanks!